From 906811c59fd76849141bbf70777ae3b79e790f79 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Sun, 5 May 2024 23:43:31 +0200
Subject: [PATCH 0840/1695] plat-stm32mp1: conf: fix no-cryp/no-rng flavors

Platform stm32mp1 flavor list no_cryp induce PKA is not embedded.
TODO: don't care. Flavor list no_cryp will not probe PKA/SAES/CRYP

no_rng platform cannot enable PKA and SAES.
TODO: this should be verified at boot time (RNG dependency upon
dt_driver_get_crypto().

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fe684bd5c543cd152fad32302b33ae1c5268c0a5 plat-stm32mp1: conf: fix no-cryp/no-rng flavors]
Change-Id: I9e9815209900abe0f4190a407973c3c369561242
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/378336
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 06b6bdd7c..d5a0581ee 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -137,11 +137,14 @@ endif
 
 ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-no_cryp)),)
 $(call force,CFG_STM32_CRYP,n)
+#$(call force,CFG_STM32_PKA,n)
 $(call force,CFG_STM32_SAES,n)
 endif
 
 ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-no_rng)),)
 $(call force,CFG_HWRNG_PTA,n)
+#$(call force,CFG_STM32_PKA,n)
+#$(call force,CFG_STM32_SAES,n)
 CFG_WITH_SOFTWARE_PRNG ?= y
 endif
 
-- 
2.47.2

